crates/uv-cli/src/lib.rs

```diff
@@ -1014,6 +1014,38 @@ pub struct PipCompileArgs {
     #[arg(long, overrides_with("all_extras"), hide = true)]
     pub no_all_extras: bool,
 
+    /// Include dependencies from the specified dependency group.
+    ///
+    /// May be provided multiple times.
+    #[arg(long, conflicts_with("only_group"))]
+    pub group: Vec<GroupName>,
+
+    /// Exclude dependencies from the specified dependency group.
+    ///
+    /// May be provided multiple times.
+    #[arg(long)]
+    pub no_group: Vec<GroupName>,
+
+    /// Exclude dependencies from default groups.
+    ///
+    /// `--group` can be used to include specific groups.
+    #[arg(long, conflicts_with_all = ["no_group", "only_group"])]
+    pub no_default_groups: bool,
```

Comment:
Do we respect `default-groups` in the `uv pip` interface? I would think not (without thinking about it too hard)

Reply:
Holy hell github's UI is haunted and won't let me unselect this textbox or cancel it, and constantly scrolls me back here, even if I refresh. I'm going to submit this comment here to see if that frees me.

Reply:
Oh my god no it keeps making a new one it DEMANDS ELABORATION on this critical discussion.

Reply:
:D